File: examples/sync/bitget_public.py
import dcex


def main() -> None:
    client = dcex.bitget()

    spot_symbols = client.get_spot_symbols(product_symbol="BTC-USDT-SPOT")
    print(spot_symbols)

    orderbook = client.get_futures_orderbook(product_symbol="BTC-USDT-SWAP", limit=5)
    print(orderbook)


if __name__ == "__main__":
    main()

File: crates/dcex/examples/bitget_public.rs
use std::time::Duration;

use dcex::exchanges::bitget::BitgetClient;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let client = BitgetClient::public(Duration::from_secs(10))?;
    let response = client.get_spot_orderbook("BTC-USDT-SPOT").limit(5).await?;
    println!("{}", response.data);
    Ok(())
}
